### DISCUSSION TITLE
netbox_napalm_plugin-api

### DISCUSSION THREAD

**Discussion by misho23-23** [netbox-community/netbox#12239] [Help Wanted!]
Hello,

Python version: 3.9.13
NetBox version: 3.5-beta1

I have just upgraded NetBox to v3.5-beta1 with Netbox Napalm Plugin - https://netbox-community.github.io/netbox-napalm/
problem is associated with Napalm added tabs id Device: /dcim/devices/<ID>/status OR (lldp-neighbors, config)
when I receive error message:
```
<class 'django.urls.exceptions.NoReverseMatch'>

'netbox_napalm_plugin-api' is not a registered namespace inside 'plugins-api'
``` 
![image](https://user-images.githubusercontent.com/130065618/231764623-9ee19560-cf86-4cb0-816c-811e318dfcc8.png)

Are you able to help me how to solve problem?
Thank you!

**Comment by misho23-23** [comment]
Downgrade  Netbox v3.4.8
Upgrade Python 3.10.11

**Reply by candlerb** [reply]
> Downgrade Netbox v3.4.8

There's no supported downgrade path from a beta though, so "reinstall netbox 3.4.8" would be a better answer.  In any case, this particular plugin only works with v3.5 (beta), so it wouldn't solve the problem (that the user is trying to test the plugin).

> Upgrade Python 3.10.11

Can you explain? The draft [release notes](https://github.com/netbox-community/netbox/blob/feature/docs/release-notes/version-3.5.md) for v3.5 don't mention any increase in the minimum python version from 3.8.

**Reply by misho23-23** [reply]
> > Downgrade Netbox v3.4.8
> 
> There's no supported downgrade path from a beta though, so "reinstall netbox 3.4.8" would be a better answer. In any case, this particular plugin only works with v3.5 (beta), so it wouldn't solve the problem (that the user is trying to test the plugin).
> 
> > Upgrade Python 3.10.11
> 
> Can you explain? The draft [release notes](https://github.com/netbox-community/netbox/blob/feature/docs/release-notes/version-3.5.md) for v3.5 don't mention any increase in the minimum python version from 3.8.

Hello,

yes, as a Newbie, I started with Netbox couple of days before and tried to use new feature used in v3.5 and it's [Configuration Template Rendering](https://github.com/netbox-community/netbox/blob/feature/docs/release-notes/version-3.5.md#configuration-template-rendering-11559) with [NAPALM plugin](https://github.com/netbox-community/netbox/blob/feature/docs/release-notes/version-3.5.md#napalm-plugin-10520) but It wasn't successful so I remove all what I did and started with new installation of Netbox v3.4.8, where Napalm is working for me for Cisco IOS...

**Reply by candlerb** [reply]
Yes, but can you explain your suggestion to upgrade to Python 3.10?  I don't believe this is good advice, in the sense that I don't think it will help the OP with their problem.

Netbox v3.4.8 definitely works with python 3.8.  (And yes, napalm support is included)

Do you have some evidence as to why python 3.10 is required when using the netbox 3.5 beta and/or the napalm plugin? If it is required, then it's probably a bug (e.g. someone has used python 3.10 features without testing on 3.8) and should be fixed.

**Reply by misho23-23** [reply]
> Yes, but can you explain your suggestion to upgrade to Python 3.10? I don't believe this is good advice, in the sense that I don't think it will help the OP with their problem.
> 
> Netbox v3.4.8 definitely works with python 3.8. (And yes, napalm support is included)
> 
> Do you have some evidence as to why python 3.10 is required when using the netbox 3.5 beta and/or the napalm plugin? If it is required, then it's probably a bug (e.g. someone has used python 3.10 features without testing on 3.8) and should be fixed.

probably, it's because of this [Fix rendering config templates from a data file](https://github.com/netbox-community/netbox/issues/12115)
